### PR TITLE
Handle asynchronous errors properly (fixes a TODO)

### PR DESCRIPTION
**Pull request by tcoulter**
Not handling errors properly could cause applications to stop functioning due to a runtime exception. For instance, if for some reason the application cannot connect to the RPC client, JSON won't be able to parse the result (an empty string), which would result in a RuntimeException.

### CODE REVIEW

File: lib/web3/httpprovider.js [hunk 1]
```diff
@@ -55,8 +55,16 @@ HttpProvider.prototype.sendAsync = function (payload, callback) {
     var request = new XMLHttpRequest();
     request.onreadystatechange = function() {
         if (request.readyState === 4) {
-            // TODO: handle the error properly here!!!
-            callback(null, JSON.parse(request.responseText));
+            var result = request.responseText;
+            var error = null;
+
+            try {
+                result = JSON.parse(result);
```
Comment (frozeman) [inline lib/web3/httpprovider.js]: This should also be add to the synchronous `send` function. (modified to work for sync and the errors should be thrown)

**Comment by tcoulter** [comment]
@frozeman How's this?

**Comment by frozeman** [comment]
can you pull develop first, there seem some conflicts. Also can you please run `npm test` to see if all test pass still?

**Comment by frozeman** [comment]
Ah and you PR should go into develop not master

**Comment by debris** [comment]
merged into develop. Thanks!

**Pull request closed without merging**